homeassistant/components/mystrom/switch.py

```diff
@@ -55,38 +61,58 @@ def name(self):
     @property
     def is_on(self):
         """Return true if switch is on."""
-        return bool(self.data["relay"])
+        return bool(self.relay)
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self.mac
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self._name,
+            "manufacturer": DOMAIN,
+            "model": "plug",
+            "sw_version": self.firmware,
+        }
 
     @property
     def current_power_w(self):
         """Return the current power consumption in W."""
-        return round(self.data["power"], 2)
+        return self.consumption
 
     @property
     def available(self):
         """Could the device be accessed during the last update call."""
         return self._available
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
         """Turn the switch on."""
         try:
-            self.plug.set_relay_on()
+            await self.plug.turn_on()
         except MyStromConnectionError:
-            _LOGGER.error("No route to device: %s", self._resource)
+            _LOGGER.error("No route to myStrom plug")
 
-    def turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs):
         """Turn the switch off."""
         try:
-            self.plug.set_relay_off()
+            await self.plug.turn_off()
         except MyStromConnectionError:
-            _LOGGER.error("No route to device: %s", self._resource)
+            _LOGGER.error("No route to myStrom plug")
 
-    def update(self):
+    async def async_update(self):
         """Get the latest data from the device and update the data."""
         try:
-            self.data = self.plug.get_status()
+            await self.plug.get_state()
+            self.firmware = self.plug.firmware
+            self.consumption = self.plug.consumption
```

Comment:
Why store this as instance variables instead of just returning it as `self.plug.firmware` when needed?